sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -114,6 +114,24 @@ object MultiFileThreadPoolUtil {
   }
 }
 
+/** A factory class to get a thread pool */
+class MultiFileThreadPoolFactory {
+  private var threadPool: Option[ThreadPoolExecutor] = None
+
+  private def initThreadPool(
+      threadTag: String,
+      numThreads: Int): ThreadPoolExecutor = synchronized {
+    if (threadPool.isEmpty) {
+      threadPool = Some(MultiFileThreadPoolUtil.createThreadPool(threadTag, numThreads))
+    }
+    threadPool.get
+  }
+
+  def getThreadPool(threadTag: String, numThreads: Int): ThreadPoolExecutor = {
```

Comment:
What happens if someone calls `getThreadPool("foo", 1)` and then later someone calls `getThreadPool("bar", 2)`?  Seems like we're just ignoring the parameters completely if someone else beat us to the call which is a surprising result to the caller who requested a certain number of threads and/or a specific tag name.

Reply:
Updated the comment and the method name to try to match the actual behavior.